src/connectors/s3.rs

```diff
@@ -0,0 +1,373 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use crate::connectors::prelude::*;
+use crate::Event;
+use async_std::channel;
+use std::{env, mem};
+use value_trait::ValueAccess;
+
+use aws_sdk_s3 as s3;
+use aws_types::{credentials::Credentials, region::Region};
+use s3::model::{CompletedMultipartUpload, CompletedPart};
+use s3::Client as S3Client;
+use s3::Endpoint;
+
+const FIVEMBS: i32 = 5 * 1024 * 1024 + 100; // Some extra bytes to keep aws happy.
+
+struct S3Connector {
+    client_sender: Option<channel::Sender<S3Client>>,
+    config: S3Config,
+    endpoint: Option<http::Uri>,
+}
+
+struct S3Sink {
+    client_receiver: channel::Receiver<S3Client>,
+    bucket: String,
+    client: Option<S3Client>,
+    buffer: Vec<u8>,
+    current_key: String,
+
+    // bookeeping for multipart uploads.
+    upload_id: String,
+    part_number: i32,
+    min_part_size: i32,
```

Comment:
Make it `usize` to avoid a cast later.

```suggestion
    min_part_size: usize,
```